atoma-proxy/src/server/mod.rs

```diff
@@ -0,0 +1,220 @@
+use std::sync::Arc;
+
+use atoma_state::{types::AtomaAtomaStateManagerEvent, AtomaStateManager};
+use atoma_sui::client::AtomaSuiClient;
+use axum::{
+    extract::State,
+    http::{HeaderMap, StatusCode},
+    routing::post,
+    Json, Router,
+};
+use flume::Sender;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use tokio::{
+    net::TcpListener,
+    sync::{oneshot, RwLock},
+};
+use tracing::error;
+
+mod config;
+pub use config::AtomaServiceConfig;
+
+use crate::sui::Sui;
+
+const CHAT_COMPLETIONS_PATH: &str = "/v1/chat/completions";
+const NODE_PUBLIC_ADDRESS_REGISTRATION: &str = "/node/registration";
+
+#[derive(Clone)]
+pub struct ProxyState {
+    pub _client: Arc<RwLock<AtomaSuiClient>>,
+    pub state_manager_sender: Sender<AtomaAtomaStateManagerEvent>,
+    pub sui: Arc<RwLock<Sui>>,
+}
+
+fn check_auth(headers: &HeaderMap) -> bool {
+    if let Some(auth) = headers.get("Authorization") {
+        if let Ok(auth) = auth.to_str() {
+            if auth == "Bearer atoma" {
+                return true;
+            }
+        }
+    }
+    false
+}
+
+pub async fn chat_completions_handler(
+    State(state): State<ProxyState>,
+    headers: HeaderMap,
+    Json(payload): Json<Value>,
+) -> Result<Json<Value>, StatusCode> {
+    if !check_auth(&headers) {
+        return Err(StatusCode::UNAUTHORIZED);
+    }
+
+    // Get the model from the payload, so we can find the task.
+    let model = if let Some(model) = payload.get("model") {
+        if let Some(model) = model.as_str() {
+            model.to_string()
+        } else {
+            return Err(StatusCode::BAD_REQUEST);
+        }
+    } else {
+        return Err(StatusCode::BAD_REQUEST);
+    };
+
+    let (result_sender, result_receiver) = oneshot::channel();
+
+    state
+        .state_manager_sender
+        .send(AtomaAtomaStateManagerEvent::GetStacksForModel {
+            model: model.clone(),
+            free_compute_units: 1000,
```

Comment:
hardcoded ?

Reply:
This is the issue I have, what should be this value? It's not like the openai request has this. It can have the max output tokens. But that doesn't count input. And if it doesn't have the max output, then it depends on the config setting of the node.

Reply:
You have the value by computing the number of tokens in the request message, and adding the `max_tokens` in the request.